### DISCUSSION TITLE
专业版的防篡改问题，无法关闭，并且每次重启1panel都必须等待防篡改命令执行一遍才可访问1panel的管理页面

### DISCUSSION THREAD

**Discussion by sky121666** [1Panel-dev/1Panel#4808] [Q&A]
如果点击过开启防篡改，即使点击了关闭（已经显示关闭），但它并没有真正的关闭！！！
![image](https://github.com/1Panel-dev/1Panel/assets/82988354/21f41143-1708-407d-b351-2b01e1cc28a7)
在升级或者其他的操作时1panel只要有重启操作，使用1pctl status 查看状态，你会发现它正在给每个文件执行锁定文件的命令，
这个问题将会让每个文件执行，会导致1panel的管理页面无法访问，必须等待执行完毕才可登录进去
![a9d7b6e66fec0a22eadd2d50db85f38](https://github.com/1Panel-dev/1Panel/assets/82988354/d5d21bc2-6ecd-4e7e-a4ce-85832030440e)

最主要是 无法真正的关闭。
当前环境为虚拟机的ubuntuserver最新版

**Comment by sky121666** [comment]
![image](https://github.com/1Panel-dev/1Panel/assets/82988354/998962e2-7ced-456f-962e-093b58a3250d)
在尝试开启关闭会出现这个提示

**Comment by sky121666** [comment]
关闭防篡改后  执行此命令后就去掉了
`sudo find /opt/1panel/apps/openresty/openresty/www/sites/你的目录/index \( -type f -o -type d \) -exec chattr -i {} \;`
![image](https://github.com/1Panel-dev/1Panel/assets/82988354/5e97ffbb-2f75-4d05-83b6-d7494f1d69ec)

**Comment by sky121666** [comment]
我就知道会这样，就算你的防篡改时关闭的，只要你再次重启1panel 就会再次执行chattr +i 的命令，下班了，等有空看看开机脚本写哪了，这个bug等待官方修复，
应该是关闭后，脚本没更新？然后又执行了命令？
![1714468286248](https://github.com/1Panel-dev/1Panel/assets/82988354/ef88b7fb-9107-4e70-9ab4-2ed67f3b22b8)

**Comment by ssongliu** [comment]
感谢反馈，麻烦加我，我看看这个问题
![image](https://github.com/1Panel-dev/1Panel/assets/73214554/44ebf234-057d-43c6-bd9d-454f395ef297)

**Comment by sky121666** [comment]
![f63351db9365ae285005ba12bf151fc](https://github.com/1Panel-dev/1Panel/assets/82988354/c433f4e4-ba37-4053-a7ce-35364e6440a9)
/opt/1panel/db/xpack/xpack.db 找到了数据库文件，打开发现有一条重复的，并且是启用状态，删除后重新导入，重新打开目录并执行命令后在重启一下1panel试一下
`sudo find /opt/1panel/apps/openresty/openresty/www/sites/你的目录/index \( -type f -o -type d \) -exec chattr -i {} \;`

**Comment by sky121666** [comment]
执行`lsattr .`后发现还带有a 属性，代表着可增加，不可删除，目录去除 -a
![image](https://github.com/1Panel-dev/1Panel/assets/82988354/b6263df2-fc5a-4160-882d-985c2d1cbd3f)

`sudo find /opt/1panel/apps/openresty/openresty/www/sites/YNEX/index -type d -exec chattr -a {} +`
![image](https://github.com/1Panel-dev/1Panel/assets/82988354/45b965ca-bbcb-446c-a951-380703a5b389)

**Comment by sky121666** [comment]
重启后，果然。。。
![image](https://github.com/1Panel-dev/1Panel/assets/82988354/f06786b8-7d1f-4527-85b9-21f27f10de0d)
![image](https://github.com/1Panel-dev/1Panel/assets/82988354/111855b8-3007-4d9d-809d-4bce45bd1684)

可以了，问题就是那数据库里面多了一条数据！！！！！！！！！！！

**Comment by sky121666** [comment]
数据库代码不该使用插入去替换，应该直接查找，去做单字段的更改，而且我个人比较建议，用目录 来做查询，
